lib/graphql/schema/input_object.rb

```diff
@@ -184,21 +184,25 @@ def validate_non_null_input(input, ctx)
             end
           end
 
-          visible_arguments_map = warden.arguments(self).reduce({}) { |m, f| m[f.name] = f; m}
-
-          # Items in the input that are unexpected
-          input.each do |name, value|
-            if visible_arguments_map[name].nil?
-              result.add_problem("Field is not defined on #{self.graphql_name}", [name])
+          # Inject missing required arguments
+          required_inputs = self.arguments.reduce({}) do |m, (argument_name, argument)|
```

Comment:
`self.arguments` _can_ be expensive, for input objects that inherit arguments from their parent classes, because it merges hashes. In practice, it might not happen much, and the most likely case (no inherited arguments) is optimized to return `own_arguments`. But, the nice part about `warden.arguments(self)` here was that it was cached in any case. Not a show-stopper but thought I'd mention it in case it comes up later.